lib/legacy/zstd_v05.rs

```diff
@@ -744,52 +731,39 @@ unsafe fn FSEv05_decompress_usingDTable_generic(
     Error::corruption_detected.to_error_code()
 }
 unsafe fn FSEv05_decompress_usingDTable(
-    dst: *mut core::ffi::c_void,
-    originalSize: size_t,
-    cSrc: *const u8,
-    cSrcSize: size_t,
+    dst: &mut [u8],
+    cSrc: &[u8],
```

Comment:
did you check that `src` and `dst` cannot overlap? That may well have been true at the time. Maybe add an assert where these slices are created just in case?